src/IROperator.cpp

```diff
@@ -515,6 +515,38 @@ void match_types(Expr &a, Expr &b) {
     }
 }
 
+// Cast to the wider type of the two. Already guaranteed to leave
+// signed/unsigned on number of lanes unchanged.
+void match_bits(Expr &x, Expr &y) {
+    if (y.type().is_int() == x.type().is_int()) {
+        // The signedness matches, so do a full cast.
```

Comment:
It seems like this case is handled by the below code too, and it would be better to minimize branches like this (which reduce test coverage, and it's just more logic we need to ensure is correct). So, I'd remove this branch and just handle every case with the else branch.

Reply:
Done.